lib/matplotlib/backends/web_backend/mpl.js

```diff
@@ -459,20 +459,30 @@ mpl.figure.prototype.mouse_event = function(event, name) {
 }
 
 mpl.figure.prototype.key_event = function(event, name) {
-    /* Don't fire events just when a modifier is changed.  Modifiers are
-       sent along with other keys. */
-    if (event.keyCode >= 16 && event.keyCode <= 20) {
-        return;
-    }
 
     var value = '';
-    if (event.ctrlKey) {
+    if (event.ctrlKey && !(event.keyCode == 17)) {
         value += "ctrl+";
     }
-    if (event.altKey) {
+    if (event.altKey && !(event.keyCode == 18)) {
         value += "alt+";
     }
-    value += String.fromCharCode(event.keyCode).toLowerCase();
+    if (event.shiftKey && !(event.keyCode == 16)) {
+        value += "shift+";
+    }
+
+    if (event.keyCode == 17) {
+        value += "control";
+    }
+    else if (event.keyCode == 18) {
+        value += "alt";
+    }
+    else if (event.keyCode == 16) {
+        value += "shift";
+    }
+    else {
+        value += String.fromCharCode(event.keyCode).toLowerCase();
```

Comment:
Is this behaviour consistent with each of the QtAgg, TkAgg and GtkAgg backends? (with regards to shift?)